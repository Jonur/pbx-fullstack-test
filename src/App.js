import React, { Component } from 'react';

class App extends Component {
  componentDidMount() {
    fetch('/api/products')
      .then((res) => res.json())
      .then(data => this.setState({
        listings: data,
      }));
  }


  render() {
    return (<main data-test="product-listings">Listing page</main>);
  }
}

export default App;
