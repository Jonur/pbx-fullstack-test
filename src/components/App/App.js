import React, { Component } from 'react';
import { getProducts } from '../../utils/api';

class App extends Component {
  state = {
    listings: {},
  }

  componentDidMount() {
    getProducts()
      .then(data => this.setState({
        listings: data,
      }));
  }


  render() {
    const { listings } = this.state;

    return ((listings.length && <main data-test="product-listings">Listing page {listings.length}</main>) || <div>Loading...</div>);
  }
}

export default App;
