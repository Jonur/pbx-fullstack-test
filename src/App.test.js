import React from 'react';
import App from './App';

describe('App Component', () => {
  const wrapper = shallow(<App />);

  it('should render the App component', () => {
    expect(wrapper).toMatchSnapshot();
  });
});