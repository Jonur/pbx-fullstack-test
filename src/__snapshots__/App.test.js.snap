// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should render the App component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Listing page",
      "data-test": "product-listings",
    },
    "ref": null,
    "rendered": "Listing page",
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Listing page",
        "data-test": "product-listings",
      },
      "ref": null,
      "rendered": "Listing page",
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
